src/analysis/AnalysisClient.js

```diff
@@ -157,8 +199,11 @@ class AnalysisClient {
         that.analyzedProductsInfo.push(analyzedProduct);
       } 
     });
-    //write price analysis info for analyzed products
-    this._writeToFile("analyzed_items_info.txt");
+
+    let categorizedItems = this._filterProductsToCategories(that.analyzedProductsInfo);
```

Comment:
Seems odd that we would need to pass an instance variable to another function that also belongs to the class. Either it shouldn't be an instance variable, or it doesn't need to be passed to `_filterProductsToCategories` at all.

Reply:
Fixed here: 8a911f8